### DISCUSSION TITLE
Scale all pages of PDF file to the page that has the maximum height

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#4818] [Digital Image Processing]
Centos 8
IM 7

I have a pdf file with pages that are not equally sized. All of them vary with about 10% margin.

I want to resize all of the pages to the page size that has the maximum dimensions.

Thanks,

**Comment by snibgo** [comment]
IM is not a general-purpose PDF editor. It can rasterize PDF pages, and can embed raster images in PDF files.

In general, if you have a bunch of input raster images, you can then:
```
 ( -clone 0--1 -layers merge +repage -set option:MAXSIZE %[fx:w]x%[fx:h] +delete )
```
... to set MAXSIZE to the maximum width and height of the input images.

**Comment by Jmuccigr** [comment]
If it's a relatively small range, you might just composite the images onto a background of the maximum size. This would avoid any weird distortions due to small scaling factors. Alternatively you could just get the page images and use something like img2pdf to put them into a PDF of a fixed size.
